### ISSUE TITLE
Data frame case sensitivity is not carried over after transform applied

### ISSUE THREAD

**Issue by jesse7zhangjc** [data-forge/data-forge-ts#138]
`caseSensitive` flag was added in this [pull request](https://github.com/data-forge/data-forge-ts/pull/86) which is super helpful for my use case. However, it only works when initiate a new data frame and won't be carried over when any data frame transform is applied.
E.g. `df.reorderSeries()`
```
const df1 = new DataFrame({
  columnNames: ['col_a', 'Col_A'],
  rows: [[1, 1]],
  caseSensitive: true,
});
console.log(df1.getColumnNames());
const df2 = df1.reorderSeries(['col_a', 'Col_A']);
console.log(df2.getColumnNames());
```

Output:
```
['col_a', 'Col_A'] // df1 column names
['col_a.1', 'col_A.2'] // df2 column names
```

After digging deeper, it turns out that `caseSensitive` is only used when [`initiateColumnNames()`](https://github.com/data-forge/data-forge-ts/blob/2a6332cdb5887414301ff8fc59579161509d879d/src/lib/dataframe.ts#L2764) and is not stored in data frame object. And it is not carried over when creating a new data frame for transform output. (e.g. in [`reorderSeries`](https://github.com/data-forge/data-forge-ts/blob/2a6332cdb5887414301ff8fc59579161509d879d/src/lib/dataframe.ts#L3710-L3733), no `caseSensitive` flag added when creating new data frame). Similarly for all other transforms.

It would also be nice to have some kind of global config that can be easily configured if we want to make all data frames case sensitive by default. That would help people to switch it on and off without making huge changes to the existing code base.

**Comment by ashleydavis** [comment]
I'm accepting PRs if you can make this work the way you want!

**Comment by ashleydavis** [comment]
I've fixed this issue in 0d4d4cb7a2c60646f9ffd0cc76aaef6ff28b2c1b and published release v1.9.3

@jesse7zhangjc thanks for logging the issue. I've used your code as a test case here: https://github.com/data-forge/data-forge-ts/blob/e9f8ee22c977b91abfa9d55febcdcbe201a28d05/src/test/dataframe.series.test.ts#L837
